### ISSUE TITLE
Expose command line option to configure FileMode and DirMode

### ISSUE THREAD

**Issue by mlusetti** [restic/rest-server#189]
Output of `rest-server --version` 
---------------------------------
`rest-server version rest-server 0.11.0 compiled with go1.17.3 on linux/amd64`

What should rest-server do differently?
---------------------------------------
Expose options to let user configure the desired _FileMode_ and _DirMode_ the server use to store files

What are you trying to do? What is your use case?
-------------------------------------------------
I want to use [Syncthing.net](https://syncthing.net/) to replicate over a peer(s) backup server(s)

Did rest-server help you today? Did it make you happy in any way?
-----------------------------------------------------------------
It has helped and made me happy every day since I started using it some years ago ... Thanks!

**Comment by rawtaz** [comment]
@mlusetti Can you please explain the initial problem you're trying to solve here? Obviously it's about filesystem permissions - can you show a clear example of what the ones that are problematic for you look like, and elaborate on the issue at hand?

**Comment by mlusetti** [comment]
> @mlusetti Can you please explain the initial problem you're trying to solve here? Obviously it's about filesystem permissions - can you show a clear example of what the ones that are problematic for you look like, and elaborate on the issue at hand? The fact that you want to use Syncthing isn't something that should be impossible to do out of the box per se.

The main issue is I need or will to use different "unix" users to run rest-server and syncthing giving each on a "unix group" in common so I can give the group "read" permission on rest-server storage directory so I'm pretty sure the syncthing process will never write to rest-directory but only read even if I misconfigure it.

Thanks for your attention.

**Comment by mlusetti** [comment]
> That effectively means that it would be enough to have the option to store files and directories such that they are group accessible. Then you can use ACLs to grant syncthing access to your repository?

Yes, using ACL or just standard unix permissions. Anyway that would be enough.

**Comment by schoettl** [comment]
Is there any progress in #308 ?

I have a similar problem: I'm using rsnapshot to sync restic repositories to a remote secondary backup server. I'd also need read permissions for group.

**Comment by MichaelEischer** [comment]
Resolved via #308 and #320.
